src/profile-logic/profile-data.js

```diff
@@ -639,19 +639,48 @@ export function getTimingsForCallNodeIndex(
   return { forPath: pathTimings, forFunc: funcTimings, rootTime };
 }
 
+// This function computes the time range for a thread, using both its samples
+// and markers data.
 export function getTimeRangeForThread(
-  thread: Thread,
+  { samples, markers }: Thread,
   interval: Milliseconds
 ): StartEndRange {
-  if (thread.samples.length === 0) {
-    return { start: Infinity, end: -Infinity };
+  const result = { start: Infinity, end: -Infinity };
+
+  if (samples.length) {
+    const lastSampleIndex = samples.length - 1;
+    result.start = samples.time[0];
+    result.end = samples.time[lastSampleIndex] + interval;
   }
 
-  const lastSampleIndex = thread.samples.length - 1;
-  return {
-    start: thread.samples.time[0],
-    end: thread.samples.time[lastSampleIndex] + interval,
-  };
+  if (markers.length) {
+    // Finding start and end times sadly requires looping through all markers :(
```

Comment:
Could this be an if/else? That'd save looping through markers on every single thread.

```
if (samples.length) {
  ...
} else if (markers.length) {
  ...
}

Reply:
With this method, we'd have the markers as a fallback from not having any sample. I think it's less correct, so I'd rather keep the current solution.

Reply:
Works for me, thanks.